### PR TITLE
Move source maps to the store

### PR DESCRIPTION
**Pull request by jasonLaster**

### CODE REVIEW

File: public/js/actions/sources.js [hunk 1]
```diff
@@ -52,6 +52,30 @@ function newSource(source) {
   };
 }
 
+function loadSourceMap(generatedSource) {
+  return ({ dispatch, getState }) => {
+    let sourceMap = getSourceMap(getState(), generatedSource.id);
+    if (sourceMap) {
+      return {};
```
Comment (jlongster) [inline public/js/actions/sources.js]: you can just do `return`. The thunk doesn't have to return anything.

File: public/js/actions/sources.js [hunk 2]
```diff
@@ -52,6 +52,30 @@ function newSource(source) {
...
+    }
+
+    return dispatch({
+      type: constants.ADD_SOURCE_MAP,
```
Comment (jlongster) [inline public/js/actions/sources.js]: `LOAD_SOURCE_MAP` might be a better description of this, analogous to `LOAD_SOURCE_TEXT`.

File: public/js/actions/sources.js [hunk 3]
```diff
@@ -52,6 +52,30 @@ function newSource(source) {
...
+      source: generatedSource,
+      [PROMISE]: (async function () {
+        const sourceMapURL = getSourceMapURL(getState(), generatedSource);
+        sourceMap = await networkRequest(sourceMapURL);
+
+        setSourceMap(generatedSource, sourceMap);
+        createOriginalSources(generatedSource)
+          .forEach(s => dispatch(newSource(s)));
```
Comment (jlongster) [inline public/js/actions/sources.js]: Nice, that looks very good

**Comment by jlongster** [comment]
This is looking really nice!

File: public/js/actions/sources.js (hunk 1 above)
Comment (jasonLaster) [inline public/js/actions/sources.js]: hmm, was getting a "inconsistent return"

File: public/js/actions/sources.js (hunk 2 above)
Comment (jasonLaster) [inline public/js/actions/sources.js]: agred

Comment (jasonLaster) [inline public/js/actions/sources.js]: i'll change this in the upcoming PR

File: public/js/actions/sources.js (hunk 1 above)
Comment (jlongster) [inline public/js/actions/sources.js]: oh, really? Interesting, you can see in `selectSource` we just `return`, so not sure what was throwing that

**Pull request merged**